docs/DeveloperGuide.md

```diff
@@ -420,8 +420,52 @@ The `UI` component is also responsible for:
 
 ### Logic Component
 
+**API** : [`Logic.java`](https://github.com/AY2324S1-CS2103-T16-3/tp/blob/master/src/main/java/unicash/logic/Logic.java)
+
+Here's a (partial) class diagram of the `Logic` component:
+
+<img src="images/unicash/LogicClassDiagram.png" width="500">
+
+The sequence diagram below illustrates the interactions within the `Logic` component, taking `execute("delete 1")` API call as an example.
+
+![Interactions Inside the Logic Component for the `delete 1` Command](images/unicash/DeleteTransactionSequenceDiagram.png)
+
+<div class="callout callout-important" markdown="span" style="margin-bottom: 20px;">
+**Note:** The lifeline for `DeleteTransactionCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+</div>
+
+How the `Logic` component works:
+
+1. When `Logic` is called upon to execute a command, it is passed to an `UniCashParser` object which in turn creates a parser that matches the command if necessary (e.g., `AddTransactionCommandParser`) and uses it to parse the command.
+   1. Note that not all commands would create an additional parser such as `GetBudgetCommand`, `ClearBudgetCommand`, etc.
+2. This results in a `Command` object (more precisely, an object of one of its subclasses e.g., `AddTransactionCommand`) which is executed by the `LogicManager`.
+3. The command can communicate with the `Model` when it is executed (e.g. to delete a person).
+4. The result of the command execution is encapsulated as a `CommandResult` object which is returned back from `Logic`.
+
+Here are the other classes in `Logic` (omitted from the class diagram above) that are used for parsing a user command:
+
+<img src="images/unicash/ParserClasses.png" width="600"/>
+
+How the parsing works:
+* When called upon to parse a user command, the `UniCashParser` class creates an `XYZCommandParser` (`XYZ` is a placeholder for the specific command name e.g., `AddTransactionCommandParser`) which uses the other classes shown above to parse the user command and create a `XYZCommand` object (e.g., `AddTransactionCommand`) which the `UniCashParser` returns back as a `Command` object.
+* All `XYZCommandParser` classes (e.g., `AddTransactionCommandParser`, `DeleteTransactionCommandParser`, ...) inherit from the `Parser` interface so that they can be treated similarly where possible e.g, during testing.
+* Note that there is a dependency drawn between `UniCashParser` and `XYZCommand` as not all commands require the use of a specialised parser class.
+
 ### Model Component
 
+**API** : [`Model.java`](https://github.com/AY2324S1-CS2103-T16-3/tp/blob/master/src/main/java/unicash/model/Model.java)
+
+<img src="images/unicash/ModelClassDiagram.png" width="500">
+
+The `Model` component,
+
+* stores UniCa$h data i.e., all `Transaction` objects (which are contained in a `TransactionList` object) and `Category` objects (which are contained in a `UniqueCategoryList` object).
```

Comment:
Actually is it `Category` object stored directly inside `UniCash` or is it `Budget`? The phrasing here seems off

Reply:
Hmm, I think both is applicable. Since its the model component, I think `Category` in the `UniqueCategoryList` could deserve a mention as well. I will add in Budget though